### ISSUE TITLE
Unknown function warn at /etc/puppet/modules/freeradius/manifests/init.pp:16 on node

### ISSUE THREAD

**Issue by lmorfitt** [djjudas21/puppet-freeradius#18]
Hello,

I'm trying to use the module from puppet forge, version 0.4.3 and I am seeing the below error.

Unknown function warn at /etc/puppet/modules/freeradius/manifests/init.pp:16 on node

This is caused when control socket is enabled as per the below code.

```puppet
class { 'freeradius':
  control_socket  => true,
  max_requests    => 4096,
  max_servers     => 4096,
  mysql_support   => true,
  utils_support   => true,
}
```

This is using puppet server and client 3.7

for the time being i've disabled the controls sockets.

thanks
Luke

**Comment by djjudas21** [comment]
Seems I made a typo and `warn` should actually be `warning`. I'll fix that in a minute.

However, you are triggering the warning by trying to use a deprecated feature. Don't set:

```puppet
class { 'freeradius':
  control_socket  => true,
}
```

Instead, use the seperate control_socket class [as documented](https://github.com/djjudas21/puppet-freeradius#freeradiuscontrol_socket):

```puppet
class { 'freeradius::control_socket':
  mode => 'ro',
}
```

**Comment by djjudas21** [comment]
Fixed in 0.4.4: https://forge.puppetlabs.com/jgazeley/freeradius

**Comment by lmorfitt** [comment]
Wow, very quick. Thank you!
